### PR TITLE
Minor cleanup to the PCP platform from further valgrind testing

### PR DESCRIPTION
**Pull request by natoscott**

### CODE REVIEW

File: pcp/Platform.c [hunk 1]
```diff
@@ -370,6 +370,10 @@ void Metric_enableThreads(void) {
 }
 
 bool Metric_fetch(struct timeval *timestamp) {
+   if (pcp->result) {
+      pmFreeResult(pcp->result);
+      pcp->result = NULL;
+   }
    int sts = pmFetch(pcp->total, pcp->fetch, &pcp->result);
    if (sts < 0) {
       if (pmDebugOptions.appl0)
```
Comment (BenBE) [inline pcp/Platform.c:379]: Unnecessary extra closing paren in error message …

File: pcp/Platform.c [hunk 2]
```diff
@@ -370,6 +370,10 @@ void Metric_enableThreads(void) {
 }
 
 bool Metric_fetch(struct timeval *timestamp) {
+   if (pcp->result) {
+      pmFreeResult(pcp->result);
```
Comment (BenBE) [inline pcp/Platform.c:374]: If `pmFreeResult` is `NULL`-safe, why check for `NULL` here? If it's not, then `Platform_done` should have that check too.

File: pcp/Platform.c [hunk 3]
```diff
@@ -502,6 +506,8 @@ void Platform_init(void) {
 
 void Platform_done(void) {
    pmDestroyContext(pcp->context);
+   pmFreeResult(pcp->result);
```
Comment (BenBE) [inline pcp/Platform.c:509]: Shouldn't the result be freed before freeing the context? Which resource is dependent on which?

**Review by BenBE — CHANGES_REQUESTED** [review]

File: pcp/Platform.c (hunk 2 above)
Comment (natoscott) [reply pcp/Platform.c:374]: Yep, needs a NULL check in done too - thanks!

File: pcp/Platform.c (hunk 3 above)
Comment (natoscott) [reply pcp/Platform.c:509]: There's no dependency there - doesn't matter which happens first.

**Review by cgzones — APPROVED** [review]

**Review by BenBE — APPROVED** [review]

**Pull request merged**